textdb/textdb-textql/src/main/javacc/TextQLParser.jj

```diff
@@ -0,0 +1,697 @@
+ *      ""        = > '' (empty)
+ *      "abc"     = > 'abc'
+ *      "de f"    = > 'de f'
+ *      "de\" f"  = > 'de" f'
+ *      "d\"e\"f" = > 'd"e"f'
+ *      "de\n f"  = > 'de\n f'
+ *      "de\\f"   = > 'de\\f'
+ * @return The String representation of the STRING_LITERAL
+ */
+String stringLiteralToString():
+{
+    Token token;
+}
+{
+    token = <STRING_LITERAL>
+    {
+        // get the matched string
+        String string = token.image; 
+        // remove delimiter quotes
+        String unquotedString = string.substring(1, string.length()-1);
+        // convert escaped quotes
+        String escapedUnquotedString = unquotedString.replace("\\\"", "\"");
+        // return the computed string
+        return escapedUnquotedString;
+    }
+}
+/**
+ * Consume a REGEX_LITERAL from input and return convert it to String 
+ * Escaped slashes and delimiter slashes are removed
+ * Examples of REGEX_LITERAL and the output String(without delimiter quotes):
+ *      //         = > "" (empty)
+ *      /abc/      = > "abc"
+ *      /de f/     = > "de f"
+ *      /d\/e/     = > "d/e"
+ *      /d\n/      = > "d\n"
+ *      /d\/e\/f/  = > "d/e/f"
+ *      /"/        = > "\"" (a quote)
+ * @return The String representation of the REGEX_LITERAL
+ */
+String regexLiteralToString():
+{
+    Token token;
+}
+{
+    token = <REGEX_LITERAL>
+    {
+        // get the matched string
+        String regex = token.image;
+        // remove delimiter slashes
+        String unslashedRegex = regex.substring(1, regex.length()-1); 
+        // convert escaped slashes
+        String escapedUnslashedRegex = unslashedRegex.replace("\\/", "/");
+        // return the computed string
+        return escapedUnslashedRegex;
+    }
+}
+/**
+ * Consume an IDENTIFIER_LITERAL from input and convert it to String
+ * Examples: i, id, id0, i0d
+ * @return The String representation of the IDENTIFIER_LITERAL
+ */
+String identifierLiteralToString():
+{
+    Token token;
+}
+{
+    token = <IDENTIFIER_LITERAL>
+    {
+        // return the matched string
+        return token.image;
+    }
+}
+
+/*
+ *    List of literal tokens separated by commas
+ */
+/**
+ * Consume a list of IDENTIFIER_LITERAL (one or more) from input and convert it to List<String>
+ * Example: "i,id,id0,i0d" (without quote) = > { "i", "id", "id0", "i0d" }
+ * Grammar: <IdentifierLiteral> ("," <IdentifierLiteral>)* 
+ * @return The List<String> representation of the list of IDENTIFIER_LITERAL
+ */
+List<String> identifierListToListString():
+{
+    String identifier;
+    List<String> identifiers = new ArrayList<String>();
+}
+{
+    identifier = identifierLiteralToString() { identifiers.add(identifier); } 
+    (
+        <COMMA> identifier = identifierLiteralToString() { identifiers.add(identifier); }
+    )*
+    {
+        return identifiers;
+    }
+}
+
+
+/*
+ * Declarations of statement rules
+ */
+/**
+ * Consume the whole input and generates a list containing the declared statement
```

Comment:
"generates" -> "generate".  Should "statement" be "statements" since it's a "list"?